### ISSUE TITLE
Incorrect values from mmol/l Nightscout source 

### ISSUE THREAD

**Issue by hummelstrand** [t1dtools/GlucoseBar#3]
I'm getting weird values offset with an open Nightscout site as source. First installation, didn't want to restart the app until I know if I should capture a log or something that might help you reproduce this.

<img width="1190" alt="image" src="https://github.com/user-attachments/assets/8b5d1911-84b5-4f8b-8258-8dca1cc697eb">
<img width="619" alt="image" src="https://github.com/user-attachments/assets/e0b62c24-9fb2-40be-bc45-d501857d14b8">

**Comment by hummelstrand** [comment]
M2 MacBook Air running Sonoma 14.5.

**Comment by AndreasStokholm** [comment]
This is very strange. Can you report on the version of NS you're running? I have only tested with 15+ myself.

**Comment by hummelstrand** [comment]
15.0.2

**Comment by AndreasStokholm** [comment]
I've been doing some debugging for a bit now and have not been able to reproduce this. Would it be possible for me to try connecting to your NS instance to try this out? You know how to get a hold of me 😁

**Comment by kskandis** [comment]
I just installed your released app and am also getting incorrect values for mg/dL.  My nightscout instance in a local server using 15.0.3.

UPDATE: it is working now.  I just quit the glucoseBar app and restarted it and now it is working.  Thank you so much!  Great tool for the Mac!

<img width="357" alt="image" src="https://github.com/user-attachments/assets/e9ffd91f-628e-4083-a988-cf59829a4302">

**Comment by AndreasStokholm** [comment]
Very interesting. I'll have to dive deeper on this.

I also noticed that the time is _completely_ off on your screenshot @kskandis. Do you have the NS server in a very different TZ than your Mac?

**Comment by kskandis** [comment]
> I also noticed that the time is _completely_ off on your screenshot @kskandis. Do you have the NS server in a very different TZ than your Mac?

No, the NS instance is running on a local rPI, the same timezone as my Mac.

**Comment by DaniBedz** [comment]
I noticed this the first time I ran the app also. It's almost like it's using someone else's/some test data on the first run.

**Comment by AndreasStokholm** [comment]
Been trying to provoke this issue, but have been unable. Hoping someone else is able to reproduce and post reproducible steps here.

I'll keep this in mind if I ever see it.

**Comment by rahim** [comment]
I also just run in to this issue on first install.

I'm using:
- Sequoia 15.2
- Nightscout 14.2.6
- mmol/L
- server and client machine are both UTC I believe

I reported it in an AppStore review, not realising the app was open source (thank you!) – I'd have tried to be more useful here if I'd known.

<img width="2056" alt="Image" src="https://github.com/user-attachments/assets/e634ae13-3920-4367-92f8-f7094d294491" />

I speculated that even though I'd saved and tested Nightscout credentials successfully it might have somehow defaulted to using the Simulation?

When I shut the app and reopened it no longer showed a reading or graph data.

**Comment by tdrkDev** [comment]
+1. Glucose values are outdated for 3 days.

**Comment by AndreasStokholm** [comment]
Finally figured out what this is caused by and fixed it in a branch. Next release will have it solved :)
